src/services/completions.ts

```diff
@@ -1442,7 +1442,7 @@ function createCompletionEntry(
             && !(type.flags & TypeFlags.BooleanLike)
             && !(type.flags & TypeFlags.Union && find((type as UnionType).types, type => !!(type.flags & TypeFlags.BooleanLike)))
         ) {
-            if (type.flags & TypeFlags.StringLike || (type.flags & TypeFlags.Union && every((type as UnionType).types, type => !!(type.flags & (TypeFlags.StringLike | TypeFlags.Undefined))))) {
+            if (type.flags & TypeFlags.StringLike || (type.flags & TypeFlags.Union && every((type as UnionType).types, type => !!((type.flags & (TypeFlags.StringLike | TypeFlags.Undefined)) || (type.isIntersection() && type.types.length === 2 && (type.types[0]!.flags & TypeFlags.String) && (type.types[1]!.flags & TypeFlags.Object) && (type.types[1] as ObjectType).members?.size === 0))))) {

```

Comment:
Let’s add (with an `/** @internal */` marker) `isEmptyAnonymousObjectType` in checker.ts to the `TypeChecker` API instead. This implementation is a little different and I don’t even want to think about how that difference might be observable. Then, can you make a utility function in services/utilities.ts called `isStringAndEmptyAnonymousObjectIntersection` just so it’s clear at a glance what’s happening?

Reply:
@andrewbranch Sorry for the delay. Also noticed that test was incorrect, fixed.